### PR TITLE
Add new desktop import guide

### PR DESCRIPTION
**Pull request by JMHReif**
Added new page on menu for importing CSV files with Neo4j Desktop, plus related asciidoc file and images for the guide.

### CODE REVIEW

File: graph-platform/neo4j-etl/neo4j-etl.adoc [hunk 1]
```diff
@@ -28,160 +29,182 @@ toc::[]
...
 
-There are 4 different ways that the ETL Tool can import data to Neo4j.
+There are 2 different ways that the ETL Tool can import data to Neo4j.
 Each import method has certain requirements and advantages, which are listed below.
 
-* a. Neo4j Import - fast loader for bulk import. Requires the graph database to be shutdown for loading.
-* b. Neo4j Shell - embedded importer. Also requires graph database to be shutdown for loading.
-* c. Cypher Shell - executes Cypher statements that are generated to an editable Cypher script. Graph database can be running.
-* d. Direct Cypher - has a direct BOLT connection for data import and also creates an editable Cypher script. Graph database can be running.
+* a. Bulk Import - has a direct BOLT connection for data import and also creates an editable Cypher script. Requires the graph database to be shutdown for loading.
```
Comment (jexp) [inline graph-platform/neo4j-etl/neo4j-etl.adoc]: Bulk Import is offline import with the import tool

File: graph-platform/neo4j-etl/neo4j-etl.adoc [hunk 2]
```diff
@@ -28,160 +29,182 @@ toc::[]
...
+* b. Online Import - fast loader for bulk import. Graph database is online and active during the load.
```
Comment (jexp) [inline graph-platform/neo4j-etl/neo4j-etl.adoc]: Online import uses Cypher (LOAD CSV) via Bolt for running databases.

File: data-import/desktop-csv-import/desktop-csv-import.adoc [hunk 3]
```diff
@@ -0,0 +1,420 @@
...
+image::{img}desktop_import_folder.jpg[link="{img}desktop_import_folder.jpg",role="popup-link"]
+
+Now you can click on the other Finder window where the zip file downloaded and move the entire .zip file to the folder Neo4j Desktop opened.
+You should see something like this when you are finished.
+
+image::{img}desktop_folder_with_zip.jpg[link="{img}desktop_folder_with_zip.jpg",role="popup-link"]
+
+Double-click on the zip file, and it should unzip and create a `desktop-csv-import` folder.
+If you click on that folder, you should see 3 .csv files - products.csv, orders.csv, and order-details.csv.
+You can open each of the files to see what the data looks like.
```
Comment (jexp) [inline data-import/desktop-csv-import/desktop-csv-import.adoc:112]: show example like first two or three lines.

File: data-import/desktop-csv-import/desktop-csv-import.adoc [hunk 4]
```diff
@@ -0,0 +1,420 @@
...
+The final column is the product unit cost.
+Though the sample values from our inspection above are all whole numbers, we know that monetary amounts can have decimal place values, as well, so we will convert these values to floats using the `toFloat()` function.
+
+We can see the Cypher to handle all of these conversions below; however, we are still not loading the values into Neo4j yet.
+We are just viewing the CSV files with converted values.
+
+[source, cypher]
+----
+LOAD CSV FROM 'file:///desktop-csv-import/products.csv' AS row
+RETURN toInteger(row[0]), row[1], toFloat(row[2])
```
Comment (jexp) [inline data-import/desktop-csv-import/desktop-csv-import.adoc]: use aliases for better column names

File: data-import/desktop-csv-import/desktop-csv-import.adoc [hunk 5]
```diff
@@ -0,0 +1,420 @@
...
+
+Note that we are using collection positions (row[0], row[1], row[2]) to refer to the columns in the row.
+In a file that has no headers, this is the way to reference values in each position.
+When we look at the next two files that have headers, you will see that we can use the column name to reference the value.
+
+.Orders.csv
+The values in the orders.csv (per the column names) are for orderID, orderDate, and shipCountry.
+Again, we can evaluate the values and determine any conversions to apply.
+
+OrderID looks like it is an integer, so we can convert that using the `toInteger()` function.
```
Comment (jexp) [inline data-import/desktop-csv-import/desktop-csv-import.adoc]: I would be wary with that oftentimes numeric values as IDs might at some point have leading zeros or such.

File: data-import/desktop-csv-import/desktop-csv-import.adoc [hunk 6]
```diff
@@ -0,0 +1,420 @@
...
+If you are using a version of Neo4j prior to 3.4, the `datetime()` function will not be available.
+That function was released in 3.4 and is supported from that version forward.
+--
+
+Just as we did with the last CSV file, let us look at the results of these conversions without importing the data.
+
+[source, cypher]
+----
+LOAD CSV WITH HEADERS FROM 'file:///desktop-csv-import/orders.csv' AS row
+RETURN toInteger(row.orderID), datetime(replace(row.orderDate,' ','T')), row.shipCountry
```
Comment (jexp) [inline data-import/desktop-csv-import/desktop-csv-import.adoc]: aliases again for nicer column headers

File: data-import/desktop-csv-import/desktop-csv-import.adoc [hunk 7]
```diff
@@ -0,0 +1,420 @@
...
+LIMIT 5
+----
+
+image::{img}desktop_import_orders_convert.jpg[link="{img}desktop_import_orders_convert.jpg",role="popup-link"]
+
+There was one tricky thing with this CSV in the `orderDate` column.
+Neo4j's datetime format will accept many formats, but the time value must start with a `T`.
```
Comment (jexp) [inline data-import/desktop-csv-import/desktop-csv-import.adoc]: that's actually ISO 8601 date format:
https://en.wikipedia.org/wiki/ISO_8601

File: data-import/desktop-csv-import/desktop-csv-import.adoc [hunk 8]
```diff
@@ -0,0 +1,420 @@
...
+The quantity field in this file is a numeric value.
+We can convert this to an integer with the `toInteger()` function we have been using.
+
+The results of these conversions are in the code below.
+Remember that we still are not loading any data yet.
+
+[source, cypher]
+----
+LOAD CSV WITH HEADERS FROM 'file:///desktop-csv-import/order-details.csv' AS row
+RETURN toInteger(row.productID), toInteger(row.orderID), toInteger(row.quantity)
```
Comment (jexp) [inline data-import/desktop-csv-import/desktop-csv-import.adoc]: column aliases again

File: data-import/desktop-csv-import/desktop-csv-import.adoc [hunk 9]
```diff
@@ -0,0 +1,420 @@
...
+LIMIT 8
+----
+
+image::{img}desktop_import_details_convert.jpg[link="{img}desktop_import_details_convert.jpg",role="popup-link"]
+
+== Loading the data!
+
+Now that we have determined that the CSV file data looks ok, and we have verified how `LOAD CSV` sees the data and converted any non-string values, we are almost ready to create the data in our graph database!
+To do that, we use Cypher statements alongside the `LOAD CSV` commands we used above.
```
Comment (jexp) [inline data-import/desktop-csv-import/desktop-csv-import.adoc:279]: Cypher update statements

File: data-import/desktop-csv-import/desktop-csv-import.adoc [hunk 10]
```diff
@@ -0,0 +1,420 @@
...
+The `LOAD CSV` will read the files, and the Cypher statements will create the data in our database.
+
+=== Graph data model
+
+An important step we need before writing Cypher statements, though, is to determine what the graph structure should look like once we import our file data.
+After all, importing the data in the existing table and column data will not provide the value we want to achieve from a graph.
+To utilize the graph database fully, we need a graph data model.
+
+We do not need to analyze all the various ways we could organize the products and orders in our files, so we have provided the model we will use for this exercise below.
```
Comment (jexp) [inline data-import/desktop-csv-import/desktop-csv-import.adoc]: Depends on the intended use-case what kind of shape we want/need.

File: data-import/desktop-csv-import/desktop-csv-import.adoc [hunk 11]
```diff
@@ -0,0 +1,420 @@
...
+
+image::{img}desktop_import_data_model.jpg[link="{img}desktop_import_data_model.jpg",role="popup-link"]
+
+We have two nodes - one for a product and one for an order.
+Each of those nodes have properties that we have from our CSV files.
+For the `Product`, we have ID, name, and unit cost.
+For the `Order`, we have ID, date/time, and country where it is going.
```
Comment (jexp) [inline data-import/desktop-csv-import/desktop-csv-import.adoc:295]: we _could_ turn country into a node - why/why-not

File: data-import/desktop-csv-import/desktop-csv-import.adoc [hunk 12]
```diff
@@ -0,0 +1,420 @@
...
+
+The order-details.csv file defines the relationship between those two nodes.
+That file has the product ID, the order ID it belongs to, and the quantity of the product on the order.
+So, in our data model, this becomes our `CONTAINS` relationship between `Product` and `Order` nodes.
+We also include a property of `quantityOrdered` on the relationship because the product quantity value only exists when a product is related to an order.
```
Comment (jexp) [inline data-import/desktop-csv-import/desktop-csv-import.adoc:300]: we can remove the product and order-id as these foreign-keys are no longer needed they are represented by the relationship itself.

File: data-import/desktop-csv-import/desktop-csv-import.adoc [hunk 13]
```diff
@@ -0,0 +1,420 @@
...
+=== Avoiding duplicates and increasing performance
+
+One final thing we need to think about before we create data in the graph is ensuring values are unique and performance is efficient.
+To handle this, we can use constraints.
+Just as with other databases, constraints ensure data integrity criteria is not violated, while simultaneously indexing the property with the constraint for faster query performance.
+
+In this exercise, we will add two constraints - one for `productId` and one for `orderId`.
+This will ensure when we create a new node of each of those types or a relationship to connect them, we know the entities are unique and indexed.
+
+Good practice is to add any constraints and indexes before loading to avoid existing violations and ensure data is added to the index as it is created.
```
Comment (jexp) [inline data-import/desktop-csv-import/desktop-csv-import.adoc]: sometimes it makes sense to add secondary indexes after the import as they can be populated in parallel and then don't affect import write performance.

File: data-import/desktop-csv-import/desktop-csv-import.adoc [hunk 14]
```diff
@@ -0,0 +1,420 @@
...
+[source, cypher]
+----
+CREATE CONSTRAINT ON (p:Product) ASSERT p.id IS UNIQUE;
+CREATE CONSTRAINT ON (o:Order) ASSERT o.id IS UNIQUE;
+----
+
+=== Cypher
+
+Now we are ready to create the Cypher for creating the data in the graph!
+We will use `CREATE` statements where we are sure that we will not have duplicate rows in our CSV file.
```
Comment (jexp) [inline data-import/desktop-csv-import/desktop-csv-import.adoc]: how are we sure?

eg.. `return count(row.productId), count(distinct row.productId)`

File: data-import/desktop-csv-import/desktop-csv-import.adoc [hunk 15]
```diff
@@ -0,0 +1,420 @@
...
+For values we know exist in the database already, we will first use `MATCH` to find them before executing any updates or inserts.
```
Comment (jexp) [inline data-import/desktop-csv-import/desktop-csv-import.adoc]: or MERGE?

File: data-import/desktop-csv-import/desktop-csv-import.adoc [hunk 16]
```diff
@@ -0,0 +1,420 @@
...
+.Products
+To start, let us load the products into the graph.
+We start with our `LOAD CSV` statement from above, then we add the Cypher to create the data from the CSV into our graph model.
+We can use the `CREATE` statement, as we feel confident that we have our constraint in place, no duplicate rows in our file, and no existing data in our graph.
+The properties will be set to the converted values we handled earlier in this guide.
+
+[source, cypher]
+----
+LOAD CSV FROM 'file:///desktop-csv-import/products.csv' AS row
+CREATE (p:Product {productId: toInteger(row[0])})
```
Comment (jexp) [inline data-import/desktop-csv-import/desktop-csv-import.adoc]: I don't think this is good practice, naming the columns early and using the names later makes for more maintainable queries.

File: data-import/desktop-csv-import/desktop-csv-import.adoc [hunk 17]
```diff
@@ -0,0 +1,420 @@
...
+You can cross-check that number with the number of rows in the CSV file from earlier (77 rows in products.csv).
+You can also run a validation query to return a sample of nodes and review that the properties look accurate.
+
+[source, cypher]
+----
+//validate products loaded correctly
+MATCH (p:Product)
+RETURN p LIMIT 20
+----
+
```
Comment (jexp) [inline data-import/desktop-csv-import/desktop-csv-import.adoc:363]: try to re-run the loading, should fail with constraint errors

File: data-import/desktop-csv-import/desktop-csv-import.adoc [hunk 18]
```diff
@@ -0,0 +1,420 @@
...
+image::{img}desktop_import_verify_products.jpg[link="{img}desktop_import_verify_products.jpg",role="popup-link"]
+
+.Orders
+Next, we can load the orders.
+Again, since we feel confident our constraint will catch violations and our CSV file data has no duplicates, we can use the `CREATE` statement to insert the data more efficiently.
```
Comment (jexp) [inline data-import/desktop-csv-import/desktop-csv-import.adoc]: would be a good opportunity to show merge.

Comment (jexp) [reply data-import/desktop-csv-import/desktop-csv-import.adoc]: also for idempotent runs.

File: data-import/desktop-csv-import/desktop-csv-import.adoc [hunk 19]
```diff
@@ -0,0 +1,420 @@
...
+MATCH (o:Order)
+RETURN o LIMIT 20
+----
+
+image::{img}desktop_import_verify_orders.jpg[link="{img}desktop_import_verify_orders.jpg",role="popup-link"]
+
+.Order-details
+Last, but not least, we create the relationship between the products and the orders.
+Since we expect all of our products and all of our orders to already exist in the graph (that data should have been loaded with the last two files), then we start with `MATCH` to find the existing `Product` and `Order` nodes.
+Then, since we know relationships do not yet exist in the graph, the `CREATE` statement will add the new relationship.
```
Comment (jexp) [inline data-import/desktop-csv-import/desktop-csv-import.adoc]: For all rows where either of the two is missing, no relationship is created.

File: data-import/desktop-csv-import/desktop-csv-import.adoc [hunk 20]
```diff
@@ -0,0 +1,420 @@
...
+MATCH (o:Order {orderId: toInteger(row.orderID)})
+CREATE (o)-[rel:CONTAINS {quantityOrdered: toInteger(row.quantity)}]->(p)
+RETURN count(rel)
+----
+
+Just as we did above, we can validate the data with the query below.
+
+[source, cypher]
+----
+MATCH (p:Product)-[rel:CONTAINS]-(o:Order)
```
Comment (jexp) [inline data-import/desktop-csv-import/desktop-csv-import.adoc]: add direction

Comment (jexp) [reply data-import/desktop-csv-import/desktop-csv-import.adoc]: why do you flip the direction here?

File: data-import/desktop-csv-import/desktop-csv-import.adoc [hunk 21]
```diff
@@ -0,0 +1,420 @@
...
+RETURN p, rel, o LIMIT 50
+----
+
+image::{img}desktop_import_verify_details.jpg[link="{img}desktop_import_verify_details.jpg",role="popup-link"]
+
+== Wrapping up
```
Comment (jexp) [inline data-import/desktop-csv-import/desktop-csv-import.adoc:423]: Should we talk about any uses of the data? or perhaps point to the next guide?

File: graph-platform/neo4j-etl/neo4j-etl.adoc [hunk 22]
```diff
@@ -28,160 +29,182 @@ toc::[]
...
 
-There are 4 different ways that the ETL Tool can import data to Neo4j.
+There are 2 different ways that the ETL Tool can import data to Neo4j.
 Each import method has certain requirements and advantages, which are listed below.
 
-* a. Neo4j Import - fast loader for bulk import. Requires the graph database to be shutdown for loading.
-* b. Neo4j Shell - embedded importer. Also requires graph database to be shutdown for loading.
-* c. Cypher Shell - executes Cypher statements that are generated to an editable Cypher script. Graph database can be running.
-* d. Direct Cypher - has a direct BOLT connection for data import and also creates an editable Cypher script. Graph database can be running.
+* a. Bulk Import - fast loader for bulk import. Requires the graph database to be shutdown for loading.
```
Comment (jexp) [inline graph-platform/neo4j-etl/neo4j-etl.adoc:186]: shut down

**Review by jexp — CHANGES_REQUESTED** [review]
In general I find that the import guide might be a bit too verbose (more blog post style than guide style) in language.
As todays users have very short attention span, they might  stop reading after the first few paragraphs. 
Perhaps you can condense some of the surrounding phrases.

Did you consider recording an accompanying video to embed into the page?

**Comment by ryguyrg** [comment]
> In general I find that the import guide might be a bit too verbose (more blog post style than guide style) in language.

I think the verbosity is all with the intent of educating users about LOAD CSV.  There are a handful of words that can be cut, but otherwise I think this is great for a tutorial.  If users want less verbosity, they should red the reference docs.  However, this step-by-step tutorial is really important for many types of users.

IMO, the non-temporal version of a blog post guiding you through something is the same thing as what should be a guide on our website.  

> As todays users have very short attention span, they might stop reading after the first few paragraphs.

Agreed-- but this isn't for those users.  They have the reference guides if they can understand how to do these things based on syntax.

> Perhaps you can condense some of the surrounding phrases.

Certainly, there are a few things to condense-- but many are educational IMO.

> 
> Did you consider recording an accompanying video to embed into the page?

Agree we should have a video guiding users through this.   Jennifer or I can take that on.

File: data-import/desktop-csv-import/desktop-csv-import.adoc (hunk 3 above)
Comment (JMHReif) [reply data-import/desktop-csv-import/desktop-csv-import.adoc:112]: Do you think that is needed when we look at it with Cypher in a few paragraphs? I'm not sure showing the files here is necessary when we review them with Cypher later (and, depending on the program they use to open the files, they could look completely different).

File: data-import/desktop-csv-import/desktop-csv-import.adoc (hunk 5 above)
Comment (JMHReif) [reply data-import/desktop-csv-import/desktop-csv-import.adoc]: Ahhhh, so would string be better for that?

Comment (JMHReif) [reply data-import/desktop-csv-import/desktop-csv-import.adoc]: Should we plan for a scenario that doesn't currently exist, though? The numbers increase sequentially and don't resemble generic number ids (more likely chance for leading zeros).

File: data-import/desktop-csv-import/desktop-csv-import.adoc (hunk 4 above)
Comment (JMHReif) [reply data-import/desktop-csv-import/desktop-csv-import.adoc]: Great point. Fixing.

File: data-import/desktop-csv-import/desktop-csv-import.adoc (hunk 6 above)
Comment (JMHReif) [reply data-import/desktop-csv-import/desktop-csv-import.adoc]: Yep, done!

File: data-import/desktop-csv-import/desktop-csv-import.adoc (hunk 7 above)
Comment (JMHReif) [reply data-import/desktop-csv-import/desktop-csv-import.adoc]: Ah, thank you! I'll reword that sentence.

File: data-import/desktop-csv-import/desktop-csv-import.adoc (hunk 8 above)
Comment (JMHReif) [reply data-import/desktop-csv-import/desktop-csv-import.adoc]: Corrected! :)

File: data-import/desktop-csv-import/desktop-csv-import.adoc (hunk 9 above)
Comment (JMHReif) [reply data-import/desktop-csv-import/desktop-csv-import.adoc:279]: We haven't loaded the data yet, though. What are we updating?

File: data-import/desktop-csv-import/desktop-csv-import.adoc (hunk 10 above)
Comment (JMHReif) [reply data-import/desktop-csv-import/desktop-csv-import.adoc]: What would be the use case for using a graph as a relational data store, though (or document or any model)? Or am I misunderstanding?

File: data-import/desktop-csv-import/desktop-csv-import.adoc (hunk 11 above)
Comment (JMHReif) [reply data-import/desktop-csv-import/desktop-csv-import.adoc:295]: That's a debate that would take time to explain fully, which might be better suited for a data modeling guide. I definitely agree that this model is very narrow in scope, but I think it's required to focus solely on import and leave other topics for other guides. Do you think that's a reasonable approach?

File: data-import/desktop-csv-import/desktop-csv-import.adoc (hunk 12 above)
Comment (JMHReif) [reply data-import/desktop-csv-import/desktop-csv-import.adoc:300]: Do you mean removing them entirely from the graph model (not even insert those values)? You would need the ids in the CSVs to know which nodes link to which, right? I'm not sure I understand.

File: data-import/desktop-csv-import/desktop-csv-import.adoc (hunk 13 above)
Comment (JMHReif) [reply data-import/desktop-csv-import/desktop-csv-import.adoc]: Ok, I can see that. I'll adjust this.

File: data-import/desktop-csv-import/desktop-csv-import.adoc (hunk 14 above)
Comment (JMHReif) [reply data-import/desktop-csv-import/desktop-csv-import.adoc]: Should we add this in the CSV import guide and not this tutorial, though? I feel this isn't information they must have to complete this exercise. If they're complete beginners, I don't want to stray into data cleanup and verification. We could make that the focus of another entire guide.

File: data-import/desktop-csv-import/desktop-csv-import.adoc (hunk 15 above)
Comment (JMHReif) [reply data-import/desktop-csv-import/desktop-csv-import.adoc]: I'm still debating between CREATE and MERGE for this guide. I think whatever way we recommend users start with is what we should use here. Is MERGE the recommended approach for all data import (or just data that we are not sure is unique/clean)? Some people seem to go one way and some another. I think we should go with 1 general approach here and save presenting other options for another guide.

File: data-import/desktop-csv-import/desktop-csv-import.adoc (hunk 16 above)
Comment (JMHReif) [reply data-import/desktop-csv-import/desktop-csv-import.adoc]: Sorry about that. I'll fix it!

File: data-import/desktop-csv-import/desktop-csv-import.adoc (hunk 17 above)
Comment (JMHReif) [reply data-import/desktop-csv-import/desktop-csv-import.adoc:363]: I'll add this check! Thanks. :)

File: data-import/desktop-csv-import/desktop-csv-import.adoc (hunk 20 above)
Comment (JMHReif) [reply data-import/desktop-csv-import/desktop-csv-import.adoc]: Oops, sorry! I'll correct.

File: data-import/desktop-csv-import/desktop-csv-import.adoc (hunk 21 above)
Comment (JMHReif) [reply data-import/desktop-csv-import/desktop-csv-import.adoc:423]: Pointing to the next guide might be a good idea! Maybe the CSV import guide to learn more details and use cases with LOAD CSV.

File: data-import/desktop-csv-import/desktop-csv-import.adoc (hunk 18 above)
Comment (JMHReif) [reply data-import/desktop-csv-import/desktop-csv-import.adoc]: Done. Thanks for suggesting this!

File: data-import/desktop-csv-import/desktop-csv-import.adoc (hunk 19 above)
Comment (JMHReif) [reply data-import/desktop-csv-import/desktop-csv-import.adoc]: Not sure you would want that if you were importing business data from another system. You wouldn't want it to arbitrarily create and Order or Product that wasn't in your relational table to begin with.....at least I wouldn't think so. Do you disagree?

**Pull request merged**